### PR TITLE
MAM-4076-2xtab-scrollup-when-all-content-is-loaded

### PR DESCRIPTION
**Pull request by bnolens**
https://linear.app/theblvd/issue/MAM-4076/2xtab-scrollup-when-all-content-is-loaded

- Tab 2xtap behavior improvements
- Feed scroll performance improvements

TODO:
- Re-enable blurhashs (but profile first and make sure the cost is reasonable)

### CODE REVIEW

**Review by derspyy — APPROVED** [review]
looks good : )

**Review by derspyy — CHANGES_REQUESTED** [review]
addressed some concerns.

File: Mammoth/Services/Backend/MastodonKit/Requests/Timelines.swift
[the remark this answers is not in the document]
Comment (bnolens) [reply Mammoth/Services/Backend/MastodonKit/Requests/Timelines.swift:1]: Looks like moth.social can return 120 posts (might be more even). So why not ask for more? If the server caps it at 40, we're still good.

[the remark this answers is not in the document]
Comment (bnolens) [reply Mammoth/Services/Backend/MastodonKit/Requests/Timelines.swift:1]: That's a good point. There was an issue with the current logic. The limit was ignored when passing a limit inside the `max` or `min` case, and the default (20) was used. I changed the code so the limit inside these cases is considered. https://github.com/TheBLVD/mammoth/pull/537/files#diff-b0738e41b5a48946cba9e561892ab40316b1f6254f651758fabdec4dfa6f4027R105

**Comment by bnolens** [comment]
Going to merge this PR. Feel free to comment on it, and I'll address issues in separate PRs

**Pull request merged**